core/templates/services/promo-bar-backend-api.service.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service Promo bar.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { ServicesConstants } from 'services/services.constants';
+
+export interface PromoBarData {
+  promoBarEnabled: boolean;
+  promoBarMessage: string;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PromoBarBackendApiService {
+  constructor(
+    private http: HttpClient) { }
+  getPromoBarData(): Promise<PromoBarData> {
+    var promoBarData = {
+      promoBarEnabled: false,
+      promoBarMessage: ''
+    };
+    if (ServicesConstants.ENABLE_PROMO_BAR) {
+      return new Promise((resolve, reject) => {
+        this.http.get<PromoBarData>(
+          ServicesConstants.PROMO_BAR_URL, {}
+        ).toPromise().then(response => {
+          resolve(response);
+        }, errorResponse => {
+          reject(errorResponse.error.error);
+        });
+      });
+    } else {
+      return new Promise((resolve, reject) => {
+        resolve(promoBarData);
+      });
+    }
```

Comment:
```suggestion
    if (!ServicesConstants.ENABLE_PROMO_BAR) {
      return new Promise((resolve, reject) => {
        resolve(promoBarData);
      });
    }
    return new Promise((resolve, reject) => {
      this.http.get<PromoBarData>(
        ServicesConstants.PROMO_BAR_URL, {}
      ).toPromise().then(response => {
        resolve(response);
      }, errorResponse => {
        reject(errorResponse.error.error);
      });
    });
 
```
Optional*
Following these patterns:

1. Complete the function as quickly as possible.
2. Avoid having multiple branching.

Both of the above patterns make it easier to read the code.

Reply:
Done!